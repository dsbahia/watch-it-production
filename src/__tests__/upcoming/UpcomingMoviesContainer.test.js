import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UpcomingMoviesContainer from "../../components/Upcoming/UpcomingMoviesContainer";

describe("UpcomingMoviesContainer", () => {
  it("should render without errors", () => {
    render(
      <MemoryRouter>
        <UpcomingMoviesContainer />
      </MemoryRouter>,
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UpcomingMoviesContainer />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a list of upcoming movies", async () => {
    render(
      <MemoryRouter>
        <UpcomingMoviesContainer />
      </MemoryRouter>,
    );

    const movieList = await screen.findByTestId("movie-list");

    expect(movieList).toBeInTheDocument();
  });
});
