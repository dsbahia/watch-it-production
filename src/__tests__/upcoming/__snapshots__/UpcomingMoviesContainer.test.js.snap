// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingMoviesContainer matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/"
    >
      <img
        alt="Watch It Logo"
        class="watch-it-logo"
        src="watch-it.png"
      />
    </a>
    <div
      class="top-rated-container"
      data-testid="movie-list"
    >
      <div
        class="top-rated-movie-title"
      >
        Upcoming Movies
      </div>
      <div
        class="top-rated-results"
      />
       
    </div>
  </div>
</DocumentFragment>
`;
